CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+add_library (NIH::lz4 ALIAS lz4)
+target_link_libraries (ripple_libs INTERFACE NIH::lz4)
+## pseudo-install our files so that dependent builds can find them
+file (MAKE_DIRECTORY
+    ${CMAKE_BINARY_DIR}/lz4/include
+    ${CMAKE_BINARY_DIR}/lz4/lib)
+target_include_directories (lz4 PUBLIC ${CMAKE_BINARY_DIR}/lz4/include)
+add_custom_command (TARGET lz4 POST_BUILD
+  COMMAND ${CMAKE_COMMAND} -E copy
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/lz4/lib/lz4.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/lz4/lib/lz4frame.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/lz4/lib/lz4hc.h
+    ${CMAKE_BINARY_DIR}/lz4/include
+  COMMAND ${CMAKE_COMMAND} -E copy
+    $<TARGET_FILE:lz4> ${CMAKE_BINARY_DIR}/lz4/lib)
+
+#[===================================================================[
+   NIH dep: sqlite
+#]===================================================================]
+
+add_library (sqlite3 STATIC
+  src/sqlite/sqlite/sqlite3.c)
+#[=========================================================[
+   When compiled with SQLITE_THREADSAFE=1, SQLite operates
+   in serialized mode. In this mode, SQLite can be safely
+   used by multiple threads with no restriction.
+
+   VFALCO NOTE: This implies a global mutex!
+
+   When compiled with SQLITE_THREADSAFE=2, SQLite can be
+   used in a multithreaded program so long as no two
+   threads attempt to use the same database connection at
+   the same time.
+
+   VFALCO NOTE: This is the preferred threading model.
+#]=========================================================]
+target_compile_definitions (sqlite3
+  PRIVATE
+    SQLITE_THREADSAFE=1
+    HAVE_USLEEP=1)
+target_compile_options (sqlite3
+  PRIVATE
+    $<$<BOOL:${MSVC}>:
+      -wd4100
+      -wd4127
+      -wd4232
+      -wd4244
+      -wd4701
+      -wd4706
+      -wd4996
+    >
+    $<$<NOT:$<BOOL:${MSVC}>>:-Wno-array-bounds>)
+add_library (NIH::sqlite3 ALIAS sqlite3)
+target_link_libraries (sqlite3 PUBLIC $<$<NOT:$<BOOL:${MSVC}>>:dl>)
+target_link_libraries (ripple_libs INTERFACE NIH::sqlite3)
+file (MAKE_DIRECTORY
+    ${CMAKE_BINARY_DIR}/sqlite3/include
+    ${CMAKE_BINARY_DIR}/sqlite3/lib)
+target_include_directories (sqlite3
+  PRIVATE
+    src/sqlite
+    src/sqlite/sqlite
+  INTERFACE
+    ${CMAKE_BINARY_DIR}/sqlite3/include)
+add_custom_command (TARGET sqlite3 POST_BUILD
+  COMMAND ${CMAKE_COMMAND} -E copy_directory
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/sqlite/sqlite ${CMAKE_BINARY_DIR}/sqlite3/include
+  COMMAND ${CMAKE_COMMAND} -E copy
+    $<TARGET_FILE:sqlite3> ${CMAKE_BINARY_DIR}/sqlite3/lib
+  BYPRODUCTS ${CMAKE_BINARY_DIR}/sqlite3/include/sqlite3.h)
+
+#[===================================================================[
+   NIH dep: soci
+#]===================================================================]
+
+add_library (soci STATIC
+  #[=========================================================[
+     this unity file is our interpretation of
+     what sources comprise soci-lib.
+     (this must be investigated/confirmed whenever updated)
```

Comment:
Did you investigate/confirm this since you're the lastest to update it by porting it to modern CMake? ;)

Reply:
I think comment is confusing and I'll remove it. What I meant is that we (at some point) decided we knew what files constitute the SOCI library and we made a unity file to represent this. Whenever you update the underlying project, it's possible files could be added/renamed/removed/etc. and thus the unity file should be reviewed for accuracy. I plan in the near future to make soci and rocksdb external projects and get out of the business of deciding what files (and settings) are needed for these projects -- they have their own build systems to make these decisions.